swimlane/core/resources/app.py

```diff
@@ -33,8 +33,22 @@ def __init__(self, swimlane, raw):
         self.id = self._raw['id']
         self.tracking_id = self._raw.get('trackingFieldId')
 
-        self._fields_by_id = {f['id']: f for f in self._raw['fields']}
-        self._fields_by_name = {f['name']: f for f in self._raw['fields']}
+        self._fields_by_id = dict()
+        self._fields_by_name = dict()
+        self._defaults = dict()
+
+        for f in self._raw['fields']:
```

Comment:
we need to give variable meaningful names, what's ok in comprehension (same line) is loosing it's context quick 8 lines down. Please change the letters to full varialbe name, v => list_values etc